zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java

```diff
@@ -52,7 +75,7 @@ public static Either<ProblemDetail, List<ProcessInstanceItem>> toProcessInstance
             .toList());
   }
 
-  public static Either<ProblemDetail, ProcessInstanceItem> toProcessInstance(
+  private static Either<ProblemDetail, ProcessInstanceItem> toProcessInstance(
```

Comment:
🔧 We should only work with `Either` if it makes sense to keep the code simple.

That could be in the following cases:
1. The result can be either a success or failure (e.g. if validation takes place)
2. The specific method is used as a reference in a more generic call/structure and one of the other methods referenced could be a success or failure (e.g. `SearchQueryRequestMapper::toProcessInstanceFilter` that is used as a method reference and does itself not need the `Either` return but other filter methods might, so it's convenient to use it there).

I don't see any of the cases fulfilled here for `toProcessInstances` and `toProcessInstance`, also not for `toDecisionDefinitions` and `toDecisionDefinition`. There might be more of those here.

Please correct me if I overlooked anything.

Reply:
I had the same thoughts when I saw that it is being used in `toProcessInstances`. I agree there is no reason to use it. I don't see a case where we would get errors when mapping the results to the DTO